lib/cog/command/alias_expander.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Cog.Command.AliasExpander do
+  alias Cog.Repo
+  alias Cog.Models.UserCommandAlias
+  alias Cog.Models.SiteCommandAlias
+  alias Cog.Command.CommandResolver
+  alias Piper.Command.Parser
+  alias Piper.Command.Ast
+
+  @moduledoc """
+  Expands aliases into pipelines
+  """
+
+  @doc """
+  expand takes a fully qualified alias name and returns the pipeline cooresponding
+  to the associated command string or an error if an alias cannot be found.
+  """
+  def expand(alias_name) do
+    limit = Application.get_env(:cog, :max_alias_expansion, 5)
+    case is_alias?(alias_name) do
+      true ->
+        expand(alias_name, [], limit, 0)
+      false ->
+        {:ok, :not_an_alias}
+    end
+  end
+
+  defp expand(alias_name, acc, limit, current_depth) do
+    cond do
```

Comment:
You should be able to replace this `cond` with pattern matching in a guard: `when current_dept <= limit`